further_link/util/bluetooth/server.py

```diff
@@ -1,56 +1,101 @@
 import asyncio
 import logging
 from os import environ
-from typing import List, Optional
+from typing import Optional
 
 from bluez_peripheral.advert import Advertisement
 from bluez_peripheral.agent import NoIoAgent
 from bluez_peripheral.gatt.service import Service
 from bluez_peripheral.util import Adapter, get_message_bus
 
+from further_link.util.bluetooth.dis_service import DeviceInformationService
 from further_link.util.bluetooth.service import FurtherGattService
-from further_link.util.bluetooth.utils import get_bluetooth_server_name
+from further_link.util.bluetooth.utils import (
+    find_object_with_uuid,
+    get_bluetooth_server_name,
+)
 
 
 class BluetoothServer:
-    _services_cls = [FurtherGattService]
-    services: List
+    _services_cls = [DeviceInformationService, FurtherGattService]
 
     def __init__(self) -> None:
-        self.services = [service() for service in self._services_cls]
+        self._adapter = None
+        self._next_service_id = 0
+        self.services = []
         self.bus = None
 
+        # Initialize services with unique paths
+        for service_cls in self._services_cls:
+            if service_cls == DeviceInformationService:
+                # Use a path that should take precedence to avoid conflicts with BlueZ's DIS
+                service_path = "/org/bluez/hci0/service_dis"
+            else:
+                service_path = (
+                    f"/com/spacecheese/bluez_peripheral/service{self._next_service_id}"
+                )
+                self._next_service_id += 1
+
+            service_instance = service_cls()
+            if hasattr(service_instance, "set_path"):
+                service_instance.set_path(service_path)
+            service_instance._path = service_path
+            self.services.append(service_instance)
+
     def get_service(self, uuid: str) -> Optional[Service]:
-        for service in self.services:
-            if service.UUID == uuid:
-                return service
-        return None
+        return find_object_with_uuid(self.services, uuid)
+
+    async def _configure_gap(self):
```

Comment:
we don't think this is necessary and it's odd that server name and appearance are duplicated with the advertisement.

do we keep it?